satpy/readers/seviri_base.py

```diff
@@ -373,7 +374,8 @@ def _tl15(self, data, wavenumber):
 
     def _vis_calibrate(self, data, solar_irradiance):
         """Calibrate to reflectance."""
-        return data * 100.0 / solar_irradiance
+        reflectance = np.pi * data * 100.0 / solar_irradiance
```

Comment:
Would it be more clear to apply the correction to the solar irradiance and then use the corrected irradiance to compute the reflectance?

Reply:
I thought about that, and it might be slightly more clear but would also involve passing more data between functions - the `solar_irradiance` doesn't have the date as an attribute, so reflectance would also have to be passed.
Therefore I would prefer to leave as it is now so that the code is cleaner. If there's a strong preference for correcting the irradiance then I can change it, though.

Reply:
But now you have this nice helper method `get_array_date` which can be used here, right? For example

```python
def _vis_calibrate(self, data, solar_irradiance):
    date = get_array_date(data)
    solar_irradiance_corr = apply_sun_earth_distance_corr(solar_irradiance, date)
    reflectance = np.pi * data * 100.0 / solar_irradiance_corr
    reflectance.attrs.update({...})

def apply_sun_earth_distance_corr(solar_irradiance, utc_date):
    from pyorbital.astronomy import sun_earth_distance_correction
    sun_earth_dist = sun_earth_distance_correction(utc_date)
    return solar_irradiance / (sun_earth_dist * sun_earth_dist)
```

Reply:
Decided not to do this, because we want `apply_sun_earth_distance_corr` to update the attributes _and_ be importable by other readers.